test/NuGet.Core.Tests/NuGet.Credentials.Test/CredentialServiceTests.cs

```diff
@@ -195,11 +195,68 @@ public async Task GetCredentialsAsync_SecondCallHasRetryTrue()
                 /*isRetry*/ false,
                 /*nonInteractive*/ false,
                 CancellationToken.None));
+
+            //cache is used instead
             _mockProvider.Verify(x => x.GetAsync(
                 uri1,
                 webProxy,
                 /*type*/ CredentialRequestType.Unauthorized,
                 /*message*/ null,
+                /*isRetry*/ false,
+                /*nonInteractive*/ false,
+                CancellationToken.None), Times.Never);
+        }
+
+        [Fact]
+        public async Task GetCredentialsAsync_SecondCallHasRetryTrue_WhenTypeIsForbidden()
```

Comment:
I'd flip the 2nd and 3rd part of the name here. https://github.com/NuGet/NuGet.Client/blob/dev/docs/coding-guidelines.md#unit-test-method-naming